pymtl3/dsl/NamedObject.py

```diff
@@ -72,25 +75,41 @@ def __setattr_for_elaborate__( s, name, obj ):
       while stack:
         u, indices = stack.pop()
 
-        if   isinstance( u, NamedObject ):
+        if isinstance( u, NamedObject ):
           # try:
             u._dsl.parent_obj  = s
             u._dsl.level       = s._dsl.level + 1
             u._dsl.my_name = u_name = name + "".join([ "[{}]".format(x)
                                                       for x in indices ])
-            u._dsl.param_dict = { None:{} }
-
-            if u_name in s._dsl.param_dict:
-              u._dsl.param_dict.update( s._dsl.param_dict[ u_name ] )
-
-            for pattern, (compiled_re, subdict) in \
-                s._dsl.param_dict[ None ].iteritems():
-              if compiled_re.match( u_name ):
-                for x, y in subdict.iteritems(): # to merge two None subdicts
-                  if x is None:
-                    u._dsl.param_dict[ None ].update( subdict )
-                  else:
-                    u._dsl.param_dict[ x ] = y
+            u._dsl.param_dict = ord_dict()
+
+            # for pattern, (compiled_re, subdict) in \
+            #     s._dsl.param_dict[ None ].iteritems():
+            #   if compiled_re.match( u_name ):
+            #     for x, y in subdict.iteritems(): # to merge two None subdicts
+            #       if x is None:
+            #         u._dsl.param_dict[ None ].update( subdict )
+            #       else:
+            #         u._dsl.param_dict[ x ] = y
+            
+            # Iterate through the param_dict and update u's dict
+            for key, subdict in s._dsl.param_dict.iteritems():
```

Comment:
Are you going to change this to the new algorithm we discussed yesterday?

Reply:
Both this and the new algorithm we discussed yesterday still won't work for some corner cases... I just re-implemented the whole thing again.